src/backends/pyplot.jl

```diff
@@ -178,14 +178,19 @@ py_fillstyle(::Nothing) = nothing
 py_fillstyle(fillstyle::Symbol) = string(fillstyle)
 
 function py_get_matching_math_font(parent_fontfamily)
-    math_font_family = if parent_fontfamily == "sans-serif"
-        "dejavusans"
-    elseif parent_fontfamily == "serif"
-        "dejavuserif"
-    else
-        parent_fontfamily
-    end
-    return math_font_family
+    # matplotlib supported math fonts according to
+    # https://matplotlib.org/stable/tutorials/text/mathtext.html
+    py_math_supported_fonts = Dict{String, String}(
+        "sans-serif" => "dejavusans", 
+        "serif" => "dejavuserif", 
+        "cm" => "cm", 
+        "stix" => "stix", 
+        "stixsans" => "stixsans"
+    )
+    # Fallback to "dejavusans" or "dejavuserif" in case the parentfont is different
+    # from supported by matplotlib fonts
+    matching_font(font) = occursin("serif", lowercase(font)) ? "dejavuserif" : "dejavusans"
+    return get(py_math_supported_fonts, parent_fontfamily, matching_font(parent_fontfamily))
 end
```

Comment:
@t-bltg I think the extrakw could be an overkill here. Only 5 mathfonts are supported by matplotlib for math fonts. I think it'd be better to check if the parentfont matches it, if not try to get a matching font based on "serif" substring

For example,
`plot(rand(10), xlabel =raw"abc $\Gamma$", fontfamily="Noto Serif")` will select "dejavuserif" for the math font. I feel this is good enough for almost everyone.

Reply:
We could update this dic further if some users prefer better font matching, i.e. create new entries